### ISSUE TITLE
Use dedicated Azure EventHubs Namespace for pre-master-kyma-gardener-azure-integration

### ISSUE THREAD

**Issue by nachtmaar** [kyma-project/kyma#8699]
**Description**
TL;DR
We are using the same Azure EventHubs Namespace in `pre-master-kyma-gardener-azure-integration`(pre-submit job) as well as in `kyma-integration-gardener-azure` (periodic job). This can cause race conditions and should be avoided.

The [gardener_azure_job_template](https://github.com/kyma-project/test-infra/blob/302a5feb5f40d4a5cc864fb527a9ef88d4a966a3/prow/jobs/kyma/kyma-integration-gardener.yaml#L99) template uses `kyma-gardener-azure` as `EVENTHUB_NAMESPACE_NAME`. Both jobs are using the template, means both jobs are using the same namespace which can cause problems.

**Expected result**
Every job has it's own Namespace and ResourceGroup and can be deleted independent of each other

**Actual result**
Race conditions might happen when jobs are running simultatenously.

**Dependency**
If we create a separate ResourceGroup and Namespace for each test run, we might end up with a lot of orphaned resources if we don't fix this bug first:
https://github.com/kyma-project/kyma/issues/8700

**Cause**

#1
How did we figure out that his problem exists?
We were investigation the failure of `core-test-external-solution` inside [kyma-integration-gardener-azure periodic job](https://storage.googleapis.com/kyma-prow-logs/logs/kyma-integration-gardener-azure/1268165615114260480/build-log.txt)

Logs of failed job:
```
 --- FAIL: TestKnativeEventingKafkaChannel (45.06s)
  FAIL
  End of logs from container "tests" in pod "oct-tp-testsuite-all-2020-06-03-13-46-knative-eventing-kafka-0" in status "Failed"
  Start of logs from container "tests" in pod "oct-tp-testsuite-all-2020-06-03-13-46-knative-eventing-kafka-1" in status "Failed"
  === RUN   TestKnativeEventingKafkaChannel
  2020/06/03 13:49:20 created Kafka channel: test-kafka-channel
  2020/06/03 13:49:20 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:49:20 [0] try failed: the Kafka channel is not ready
  2020/06/03 13:49:25 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:49:25 [1] try failed: the Kafka channel is not ready
  2020/06/03 13:49:30 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:49:30 [2] try failed: the Kafka channel is not ready
  2020/06/03 13:49:35 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:49:35 [3] try failed: the Kafka channel is not ready
  2020/06/03 13:49:40 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:49:40 [4] try failed: the Kafka channel is not ready
  2020/06/03 13:49:45 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:49:45 [5] try failed: the Kafka channel is not ready
  2020/06/03 13:49:50 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:49:50 [6] try failed: the Kafka channel is not ready
  2020/06/03 13:49:55 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:49:55 [7] try failed: the Kafka channel is not ready
  2020/06/03 13:50:00 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:50:00 [8] try failed: the Kafka channel is not ready
  2020/06/03 13:50:05 found Kafka channel: test-kafka-channel with ready status: false
  2020/06/03 13:50:05 [9] try failed: the Kafka channel is not ready
  2020/06/03 13:50:05 test failed with error: All attempts fail:
  #1: the Kafka channel is not ready
  #2: the Kafka channel is not ready
  #3: the Kafka channel is not ready
  #4: the Kafka channel is not ready
  #5: the Kafka channel is not ready
  #6: the Kafka channel is not ready
  #7: the Kafka channel is not ready
  #8: the Kafka channel is not ready
  #9: the Kafka channel is not ready
  #10: the Kafka channel is not ready
  2020/06/03 13:50:05 deleted Kafka channel: test-kafka-channel
```

Debugging Kafka Channel Controller:
```
./knative-eventing/knative-eventing-kafka-channel-controller-69957888d9-m52bg/logs.txt:{"level":"error","ts":"2020-06-03T13:46:37.269Z","caller":"admin/admin_eventhub.go:138","msg":"Failed To Create EventHub - Missing Error Code","Controller":"kafka-channel-controller","error":"Put https://kyma-gardener-azure.servicebus.windows.net/test-function-2910933558.default-kne-trigger?api-version=2017-04: dial tcp: lookup kyma-gardener-azure.servicebus.windows.net on 100.64.0.10:53: no such host","stacktrace":"github.com/kyma-incubator/knative-kafka/components/controller/vendor/github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/admin.(*EventHubAdminClient).createTopic\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/admin/admin_eventhub.go:138\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/admin.(*EventHubAdminClient).CreateTopics\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/admin/admin_eventhub.go:76\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).createTopic\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/topic.go:88\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).reconcileTopic\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/topic.go:54\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).reconcile\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/reconciler.go:132\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).Reconcile\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/reconciler.go:103\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
...
377 times more the no such host error
./knative-eventing/knative-eventing-kafka-channel-controller-69957888d9-m52bg/logs.txt:{"level":"error","ts":"2020-06-03T14:23:02.104Z","caller":"admin/admin_eventhub.go:144","msg":"Failed To Create EventHub - Received Unmapped Error Code","Controller":"kafka-channel-controller","error":"error code: 401, Details: InvalidSignature: The token has an invalid signature. TrackingId:4d34319e-8756-4b63-ae3b-01a83ac7f12a_G31, SystemTracker:kyma-gardener-azure.servicebus.windows.net:test-function-1977182185.default-kne-trigger, Timestamp:2020-06-03T14:23:02","stacktrace":"github.com/kyma-incubator/knative-kafka/components/controller/vendor/github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/admin.(*EventHubAdminClient).createTopic\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/admin/admin_eventhub.go:144\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/admin.(*EventHubAdminClient).CreateTopics\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/admin/admin_eventhub.go:76\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).createTopic\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/topic.go:88\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).reconcileTopic\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/topic.go:54\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).reconcile\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/reconciler.go:132\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).Reconcile\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/reconciler.go:103\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
./knative-eventing/knative-eventing-kafka-channel-controller-69957888d9-m52bg/logs.txt:{"level":"error","ts":"2020-06-03T14:23:02.104Z","caller":"kafkachannel/topic.go:99","msg":"Failed To Create Topic (Results)","Controller":"kafka-channel-controller","Topic":"test-function-1977182185.default-kne-trigger","error":"error code: 401, Details: InvalidSignature: The token has an invalid signature. TrackingId:4d34319e-8756-4b63-ae3b-01a83ac7f12a_G31, SystemTracker:kyma-gardener-azure.servicebus.windows.net:test-function-1977182185.default-kne-trigger, Timestamp:2020-06-03T14:23:02","TopicResults":[{"Topic":"test-function-1977182185.default-kne-trigger","Error":{}}],"stacktrace":"github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).createTopic\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/topic.go:99\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).reconcileTopic\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/topic.go:54\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).reconcile\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/reconciler.go:132\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).Reconcile\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/reconciler.go:103\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
./knative-eventing/knative-eventing-kafka-channel-controller-69957888d9-m52bg/logs.txt:{"level":"error","ts":"2020-06-03T14:23:02.104Z","caller":"kafkachannel/topic.go:60","msg":"Failed To Reconcile Topic","Controller":"kafka-channel-controller","Namespace":"test-function-1977182185","Name":"default-kne-trigger","TopicName":"test-function-1977182185.default-kne-trigger","error":"test-function-1977182185.default-kne-trigger","stacktrace":"github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).reconcileTopic\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/topic.go:60\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).reconcile\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/reconciler.go:132\ngithub.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel.(*Reconciler).Reconcile\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/pkg/reconciler/kafkachannel/reconciler.go:103\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\ngithub.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/kyma-incubator/knative-kafka/components/controller/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
```

Explanation:
- the `no such host` error is caused by another job running in parallel deleting the Azure EventHubs Namespace
- `error code: 401` is probably caused by another job which created a new Azure EventHubs namespace

#2

Without digging deeper, I guess this is another cause, see [pipeline](https://storage.googleapis.com/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/8693/pre-master-kyma-gardener-azure-integration/1270376712940556289/build-log.txt):

```
#################################################################################################
# Loading the new EventHub Namespace's authorization key...
#################################################################################################
    [0m
ERROR: NotFound - The namespace kyma-gardener-azure in subscription 8526f76e07a748a390a6f62d550c11ea is not active, its current state is Created. CorrelationId: dbf9405a-585e-427f-8072-5e540c7343b7
```

**Comment by nachtmaar** [comment]
We saved the logs from the cluster as well as the resources, please ping me if you need them

**Comment by sayanh** [comment]
Examples where the the EH has names with random suffixes:
Periodic
https://storage.googleapis.com/kyma-prow-logs/logs/kyma-integration-gardener-azure/1273545216187437056/build-log.txt

Pre-submit
https://storage.googleapis.com/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/8214/pre-master-kyma-gardener-azure-integration/1273536035178942464/build-log.txt
